src/features/hostapp/hooks/target-hostapp.ts

```diff
@@ -39,6 +39,22 @@ hooks.addPureHook('PATCH', 'resin', 'device', {
 	},
 });
 
+hooks.addPureHook('PATCH', 'resin', 'device', {
+	/**
+	 * If a device changes device types, we need to clear out the related target hostapp release
+	 *
+	 * TODO: changing device types presents us with a bit of a conundrum. since we cannot guarantee parity between
+	 * available OSes for each device type, we cannot just switch the target to the new device type, so instead let's just
+	 * unset the value
+	 */
+
+	async POSTPARSE(args) {
+		if (args.request.values.is_of__device_type != null) {
+			args.request.values.should_be_operated_by__release = null;
+		}
+	},
```

Comment:
It'd be good to check that an explicit `should_be_operated_by__release` has not been set, so that someone can change the device type + pin to a new host release at the same time
```suggestion
	async POSTPARSE({ request }) {
		if (request.values.is_of__device_type != null && request.values.should_be_operated_by__release === undefined) {
			request.values.should_be_operated_by__release = null;
		}
	},
```